mmocr/models/textrecog/backbones/resnet_abi.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch.nn as nn
+from mmcv.runner import BaseModule, Sequential
+
+import mmocr.utils as utils
+from mmocr.models.builder import BACKBONES
+from mmocr.models.textrecog.layers import BasicBlock
+
+
+@BACKBONES.register_module()
+class ResNetABI(BaseModule):
+    """Implement ResNet backbone for text recognition, modified from
+      `ResNet <https://arxiv.org/pdf/1512.03385.pdf>`_ and
+      `<https://github.com/FangShancheng/ABINet>`_
+    Args:
+        base_channels (int): Number of channels of input image tensor.
+        layers (list[int]): List of BasicBlock number for each stage.
+        channels (list[int]): List of out_channels of Conv2d layer.
+        out_indices (None | Sequence[int]): Indices of output stages.
+        last_stage_pool (bool): If True, add `MaxPool2d` layer to last stage.
+    """
+
+    def __init__(self,
+                 base_channels=3,
+                 arch_settings=[3, 4, 6, 6, 3],
+                 channels=[32, 32, 64, 128, 256, 512],
+                 strides=[1, 2, 1, 2, 1, 1],
+                 out_indices=None,
+                 last_stage_pool=False,
+                 init_cfg=[
+                     dict(type='Xavier', layer='Conv2d'),
+                     dict(type='Constant', val=1, layer='BatchNorm2d')
+                 ]):
+        super().__init__(init_cfg=init_cfg)
+        assert isinstance(base_channels, int)
+        assert utils.is_type_list(arch_settings, int)
+        assert utils.is_type_list(channels, int)
+        assert utils.is_type_list(strides, int)
+        assert len(arch_settings) == len(channels) - 1
+        assert len(arch_settings) == len(strides) - 1
+        assert out_indices is None or isinstance(out_indices, (list, tuple))
+        assert isinstance(last_stage_pool, bool)
+
+        self.out_indices = out_indices
+        self.last_stage_pool = last_stage_pool
+
+        self.conv1 = nn.Conv2d(
+            base_channels,
+            channels[0],
+            kernel_size=3,
+            stride=strides[0],
+            padding=1)
+        self.bn1 = nn.BatchNorm2d(channels[0])
+        self.relu1 = nn.ReLU(inplace=True)
+
+        self.layers = Sequential(*[
+            self._make_layer(channels[i], channels[i + 1], arch_settings[i],
+                             strides[i + 1])
+            for i in range(len(arch_settings))
+        ])
+
+    def _make_layer(self, input_channels, output_channels, blocks, stride=1):
```

Comment:
Can be replaced by function?
https://github.com/open-mmlab/mmcv/blob/7207397de3c2ffaef878628a55fa2ab37ac0bbc7/mmcv/cnn/resnet.py#L143
Recommend use block=BasicBlock as parameters instead of fixed

Reply:
```suggestion
    def _make_res_layer(block,
                        inplanes,
                        planes,
                        blocks,
                        stride=1,
                        dilation=1
```
keep the same API with upstream(mmcv/mmdet)